From c59e117534ef9fdba67fd0fcb5b83b8b97abab7e Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Tue, 18 Feb 2020 14:44:15 +0200
Subject: [PATCH 64/65] try to add caos utils

	new file:   core/java/com/android/internal/util/CaosUtils.java
---
 core/java/com/android/internal/util/CaosUtils.java | 33 ++++++++++++++++++++++
 1 file changed, 33 insertions(+)
 create mode 100644 core/java/com/android/internal/util/CaosUtils.java

diff --git a/core/java/com/android/internal/util/CaosUtils.java b/core/java/com/android/internal/util/CaosUtils.java
new file mode 100644
index 00000000000..a80a23f8f84
--- /dev/null
+++ b/core/java/com/android/internal/util/CaosUtils.java
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2020 CAOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.internal.util;
+
+import android.content.Context;
+import android.os.PowerManager;
+import android.os.ServiceManager;
+import android.os.SystemClock;
+
+public class CaosUtils {
+	private static final String TAG = "CaosUtils";
+
+	public static void switchScreenOff(Context ctx) {
+		PowerManager pm = (PowerManager) ctx.getSystemService(Context.POWER_SERVICE);
+		if (pm!= null) {
+			pm.goToSleep(SystemClock.uptimeMillis());
+		}
+	}
+}
-- 
2.11.0

